### PR TITLE
[WIP] Alternative services

### PR DESCRIPTION
**Pull request by aojea**
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/filters/patch_alternative_services.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	})
+
+	a.endpointsLister = informer.Lister()
+	a.endpointsSynced = informer.Informer().HasSynced
+	return a
+}
+
+// WithAternativeServices sets the Alt-Svc header based on the available api servers
+// See RFC7838
+func WithAternativeServices(handler http.Handler, a *AlternativeServicesInfo) http.Handler {
```
Comment (dcbw) [inline staging/src/k8s.io/apiserver/pkg/server/filters/patch_alternative_services.go:115]: @aojea `WithAternativeServices` -> `WithAlternativeServices` (missing 'l')

**Comment by fedebongio** [comment]
/triage accepted
When it's ready for review/discussion please loop us in back
Thanks!

**Comment by aojea** [comment]
> /triage accepted When it's ready for review/discussion please loop us in back Thanks!

The discussion has to happen first in the kep @fedebongio :)
https://github.com/kubernetes/enhancements/issues/3037

**Comment by dims** [comment]
please reopen if you still need this WIP PR. Trying to declutter the open PR(s)

**Comment by aojea** [comment]
sure, no prob :+1:

**Pull request closed without merging**